Zend/tests/bug71572.phpt

```diff
@@ -0,0 +1,16 @@
+--TEST--
+Bug #71572: String offset assignment from an empty string inserts null byte
+--FILE--
+<?php
+
+$str = "abc";
+var_dump($str{0} = "");
+var_dump($str);
```

Comment:
Please add a test case for `$str{10} = ""` as well. Currently this will still extend the string, and likely additionally leave one uninitialized byte.